psydac/fem/partitioning.py

```diff
@@ -280,21 +300,59 @@ def construct_reduced_interface_spaces(spaces, reduced_spaces, interiors, connec
+                    max_ncells = tuple(max(ni,nj) for ni,nj in zip(Vi.ncells,Vj.ncells))
+                    Vi._refined_space[max_ncells].create_interface_space(axis_i, ext_i, cart=cart)
+                    Vj._refined_space[max_ncells].create_interface_space(axis_j, ext_j, cart=cart)
             else:
                 Vi = reduced_spaces[patch_i]
                 Vj = reduced_spaces[patch_j]
                 npts_i = [Vik.nbasis for Vik in Vi.spaces]
                 npts_j = [Vik.nbasis for Vik in Vj.spaces]
-                cart_ij         = cart_i.reduce_npts(Vi.vector_space.cart, Vj.vector_space.cart)
+                cart_ij  = cart_i.reduce_npts(Vi.vector_space.cart, Vj.vector_space.cart)
                 cart_ij.set_interface_communication_infos(get_minus_starts_ends, get_plus_starts_ends)
                 Vi.create_interface_space(axis_i, ext_i, cart=cart_ij)
                 Vj.create_interface_space(axis_j, ext_j, cart=cart_ij)
 
+                cart = InterfaceCartDecomposition(cart_ij._cart_minus, cart_ij._cart_plus,
+                                                  cart_ij._comm, [axis_i, axis_j], [ext_i, ext_j],
+                                                  [cart_ij.ranks_in_topo_minus, cart_ij.ranks_in_topo_plus],
+                                                  [cart_ij._local_group_minus, cart_ij._local_group_plus],
+                                                  [cart_ij._local_comm_minus, cart_ij._local_comm_plus],
+                                                  [cart_ij._root_rank_minus, cart_ij._root_rank_plus],
+                                                  [], reduce_elements=True)
+
+                cart.set_interface_communication_infos(get_minus_starts_ends, get_plus_starts_ends)
+                max_ncells = tuple(max(ni,nj) for ni,nj in zip(Vi.ncells,Vj.ncells))
+                Vi._refined_space[max_ncells].create_interface_space(axis_i, ext_i, cart=cart)
+                Vj._refined_space[max_ncells].create_interface_space(axis_j, ext_j, cart=cart)
         else:
-            if isinstance(reduced_spaces[patch_i], ProductFemSpace):
+            if isinstance(reduced_spaces[patch_i], VectorFemSpace):
                 for Vi,Vj in zip(reduced_spaces[patch_i].spaces, reduced_spaces[patch_j].spaces):
                     Vi.create_interface_space(axis_i, ext_i, cart=Vi.vector_space.cart)
                     Vj.create_interface_space(axis_j , ext_j , cart=Vj.vector_space.cart)
+                    max_ncells = tuple(max(ni,nj) for ni,nj in zip(Vi.ncells,Vj.ncells))
+                    cart_i = Vi._refined_space[tuple(max_ncells)].vector_space.cart
+                    cart_j = Vj._refined_space[tuple(max_ncells)].vector_space.cart
+                    Vi._refined_space[max_ncells].create_interface_space(axis_i, ext_i, cart=cart_i)
+                    Vj._refined_space[max_ncells].create_interface_space(axis_j, ext_j, cart=cart_j)
```

Comment:
See above

Reply:
done in [f74fd12](https://github.com/pyccel/psydac/pull/241/commits/f74fd122c1b075e8f616d556a9b44aa8408bb4a1)